DEVELOPERS.md

```diff
@@ -30,208 +37,242 @@ for more info. Example usage can be found [here](http://www.doxygen.nl/manual/do
+  separate functions instead of using braces. If braced scope syntax is used,
+  put `break;` statements inside the scope.
 
 ```c
 switch (n) {
 case 1:
-	break;
+        break;
 case 2: {
-	break;
+        break;
 }
 default:
 }
 ```
 
-* Lines should be at most 78 chars. A tab is considered as 8 chars.
+* Lines should be at most 78 characters in length. A tab is considered 8
+  characters.
 
 * Braces open on the same line as the for/while/if/else/function/etc. Closing
   braces are put on a line of their own, except in the else of an if statement
-  or in a while of a do-while statement. Always use braces for if and while.
+  or in the while of a do-while statement.
 
 ```c
 if (a == b) {
-	...
+        ...
 }
 
 if (a == b) {
-	...
+        ...
 } else if (a > b) {
-	...
+        ...
 }
 
 if (a == b) {
-	...
+        ...
 } else {
-	do_something_else ();
+        do_something_else ();
 }
 
 do {
-	do_something ();
+        do_something ();
 } while (cond);
 
 if (a == b) {
-	b = 3;
+        b = 3;
 }
+```
+
+* Always use braces for if and while.
 
+```diff
+-if (a == b)
+-        return;
++if (a == b) {
++        return;
++}
 ```
 
-* In general, don't use goto. The goto statement only comes in handy when a
+* In general, avoid `goto`. The `goto` statement only comes in handy when a
   function exits from multiple locations and some common work such as cleanup
   has to be done. If there is no cleanup needed, then just return directly.
 
-  Choose label names which say what the goto does or why the goto exists.  An
-  example of a good name could be "out_buffer:" if the goto frees "buffer".
-  Avoid using GW-BASIC names like "err1:" and "err2:".
+  Choose label names which say what the `goto` does or why it exists.  An
+  example of a good name could be `out_buffer:` if the `goto` frees `buffer`.
+  Avoid using GW-BASIC names like `err1:` and `err2:`.
 
-* Use `r_return_*` macros to check preconditions that are caused by
-  programmers' errors. Please, keep in mind:
-  * conditions that should never happen should be handled through
-    `r_return_*` macros;
-  * runtime conditions (e.g. malloc returns NULL, input coming from user,
-    etc.) should be handled in the usual way through if-else.
+* Use `r_return_*` macros to check for conditions that are caused by
+  programming errors or bugs; i.e.: conditions that should **never** happen. Do
+  not use them when checking for runtime error conditions, such as a `NULL`
+  value being returned from `malloc()`. Use a standard if statement for these
+  cases.
 
 ```c
 int check(RCore *c, int a, int b) {
-	r_return_val_if_fail (c, false);
-	r_return_val_if_fail (a >= 0, b >= 1, false);
-
-	if (a == 0) {
-		/* do something */
-		...
-	}
-	... /* do something else */
+        /* check for programming errors */
+        r_return_val_if_fail (c, false);
+        r_return_val_if_fail (a >= 0, b >= 1, false);
+
+        /* check for runtime errors */
+        ut8 *buf = malloc(sizeof (a) * b);
```

Comment:
```suggestion
        ut8 *buf = malloc (sizeof (a) * b);
```

Reply:
also this line exposes other bugs. there's a multiply overflow. Its better to use `calloc (sizeof(a), b)` or use the `OVFCHK` macros, which imho should be explained in this file too.

Reply:
i will add that to the to-do for the next pr